tests/test_utils/__init__.py

```diff
@@ -19,12 +21,16 @@ def compare_dicts(dict1, dict2):
         elif isinstance(value1, list) and isinstance(value2, list):
             if len(value1) != len(value2):
                 return False
-            # check if the lists contain dictionaries as elements
-            if len(value1) > 0 and isinstance(value1[0], dict) and isinstance(value2[0], dict):
-                for i in range(len(value1)):
-                    if not compare_dicts(value1[i], value2[i]):
-                        return False
-            elif sorted(value1) != sorted(value2):
+
+            # Handle lists of dictionaries (sort by keys before comparing)
+            if all(isinstance(v, dict) for v in value1) and all(isinstance(v, dict) for v in value2):
+                sorted_value1 = sorted(value1, key=lambda d: json.dumps(d, sort_keys=True))
+                sorted_value2 = sorted(value2, key=lambda d: json.dumps(d, sort_keys=True))
+                if not all(compare_dicts(d1, d2) for d1, d2 in zip(sorted_value1, sorted_value2, strict=False)):
```

Comment:
```suggestion
                if not all(compare_dicts(d1, d2) for d1, d2 in zip(sorted_value1, sorted_value2, strict=True)):
```

Only thing that's a little weird, since these should definitely be the same length

Reply:
Oops - agreed!